akka-http/src/main/scala/akka/http/scaladsl/server/RequestContextImpl.scala

```diff
@@ -109,7 +113,8 @@ private[http] class RequestContextImpl(
   /** Attempts recovering from the special case when non-2xx response is sent, yet content negotiation was unable to find a match. */
   private def attemptRecoveryFromUnacceptableResponseContentTypeException(trm: ToResponseMarshallable, supported: Set[ContentNegotiator.Alternative]): Future[RouteResult] =
     trm.value match {
-      case (status: StatusCode, value) if !status.isSuccess ⇒ this.withAcceptAll.complete(trm) // retry giving up content negotiation
```

Comment:
I think that method doesn't belong into `RequestContextImpl` at all. Let's not touch it here and I'll follow up with a better fix for https://github.com/akka/akka/issues/19397. I think we need to be really careful not to introduce any custom logic like this into central pieces of akka-http because - like here - it can be hard to reason about all the code paths flowing through this code and if it has a bug it will always affect all akka-http applications.